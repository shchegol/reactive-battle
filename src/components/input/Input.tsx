import React, { FC } from 'react';
import toClassNames from '@utils/toClassNames';

import './input.scss';
import { Props } from './types';

/**
 * Input
 * @param {string} [labelText=''] - label text
 * @param {string} [messageText=''] - help message
 * @param {boolean} [isError=false] - error boolean
 * @param rest - rest params
 * @constructor
 */

const Input: FC<Props> = ({
  labelText = '',
  messageText = '',
  isError = false,
  ...rest
}) => (
  <label // eslint-disable-line jsx-a11y/label-has-associated-control
    className={toClassNames(
      'input', {
        input_error: isError,
        input_required: rest.required,
      },
      rest.className,
    )}
  >
    <span
      className="input__label"
    >
      {labelText}
    </span>
    <input
      id={rest.id}
      type={rest.type || 'text'}
      name={rest.name}
      className="input__field"
      value={rest.value || ''}
      placeholder={rest.placeholder}
      onChange={rest.onChange}
      required={rest.required}
    />
    <span className="input__message">
      {messageText}
    </span>
  </label>
);

export default Input;
