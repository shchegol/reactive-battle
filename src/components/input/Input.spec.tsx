import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from './index';

configure({ adapter: new Adapter() });

describe('Input', () => {
  const snapshoot = (label: string, message: string, isError: boolean, onChange: () => void) => {
    const wrapper = render(<Input
      labelText={label}
      messageText={message}
      isError={isError}
      onChange={onChange}
    />);
    it('Render', () => {
      expect(wrapper).toMatchSnapshot();
    });
  };

  snapshoot('test', 'no error', false, () => { console.log('ok'); });
  snapshoot('test', 'error', true, () => { console.log('ok'); });
});
