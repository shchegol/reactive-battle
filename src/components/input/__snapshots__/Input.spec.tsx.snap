// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Render 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "input",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "input__label",
        },
        "children": Array [
          Node {
            "data": "test",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "input__field",
            "type": "text",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "input__message",
            },
            "children": Array [
              Node {
                "data": "no error",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "input__field",
          "type": "text",
          "value": "",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "input__message",
          },
          "children": Array [
            Node {
              "data": "no error",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "input__label",
          },
          "children": Array [
            Node {
              "data": "test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "input__message",
        },
        "children": Array [
          Node {
            "data": "no error",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "input__field",
            "type": "text",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "input__label",
            },
            "children": Array [
              Node {
                "data": "test",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Input Render 2`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "input input_error",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "input__label",
        },
        "children": Array [
          Node {
            "data": "test",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "input__field",
            "type": "text",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "input__message",
            },
            "children": Array [
              Node {
                "data": "error",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "input__field",
          "type": "text",
          "value": "",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "input__message",
          },
          "children": Array [
            Node {
              "data": "error",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "input__label",
          },
          "children": Array [
            Node {
              "data": "test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "input__message",
        },
        "children": Array [
          Node {
            "data": "error",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "input__field",
            "type": "text",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "input__label",
            },
            "children": Array [
              Node {
                "data": "test",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
